src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -119,15 +119,15 @@ public async Task AddDatabaseNode()
             var mentor = GetStringQueryString("mentor", false);
             var raftRequestId = GetRaftRequestIdFromQuery();
 
-            string errorMessage;
-            if (ResourceNameValidator.IsValidResourceName(name, ServerStore.Configuration.Core.DataDirectory.FullPath, out errorMessage) == false)
-                throw new BadRequestException(errorMessage);
-
             ServerStore.EnsureNotPassive();
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             using (context.OpenReadTransaction())
             {
                 var databaseRecord = ServerStore.Cluster.ReadDatabase(context, name, out var index);
+                if (databaseRecord == null)
+                {
+                    throw new InvalidOperationException("Database Record not found when trying to add a node to the database topology");
```

Comment:
I think this should be 'DatabaseDoesNotExistException' ?

Reply:
done